File: 1310-xor-queries-of-a-subarray/1310-xor-queries-of-a-subarray.java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int n = arr.length;
        int []prexor = new int[n];
        prexor[0] = arr[0];
        for(int i=1;i<n;i++){
            prexor[i] = arr[i] ^ prexor[i-1];
        }
        int ans[] = new int[queries.length];
        for(int i = 0;i<queries.length;i++){
            int l = queries[i][0];
            int r = queries[i][1];
            ans[i] = l==0 ? prexor[r]:prexor[l-1] ^ prexor[r];
        }
        return ans;
    }
}

File: 1310-xor-queries-of-a-subarray/1310-xor-queries-of-a-subarray.cpp
class Solution {
public:
     vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) 
    {   //Initialize result to 0 and size of queries
        vector<int> result(queries.size(),0);

        for(int i=0;i<queries.size();i++)   //first loop for the size of queries
        {
            result[i] = arr[queries[i][0]]; //initialize of first element, so that there is no XOR with 0
            for(int j= queries[i][0]+1; j<= queries[i][1];j++) //second loop from queries[i][0] to queries[i][1]
                result[i] ^= arr[j];    //XOR operations
        }    
        
        return result;
    }
    
};